imagemounter/disk.py

```diff
@@ -186,6 +193,16 @@ def mount(self):
             elif method == 'vmware-mount':
                 cmds.append(['vmware-mount', '-r', '-f', self.paths[0], self.mountpoint])
 
+            elif method == 'nbd':
+                self.nbd = True
+                _util.check_output_(['modprobe', 'nbd', 'max_part=63'])  # Load nbd driver
+                try:
+                    self.mountpoint = _util.get_free_nbd_device()  # Get free nbd device
```

Comment:
You are overwriting self.mountpoint, but is has been set before. This is troublesome, because this whole method has the ability to fallback to other mount methods if the first one does not succeed. This results in other methods trying to mount to an nbd device. That is obviously not the right way.

I guess we perhaps need to rethink the structure of this entire method, but now we have a temporary directory created that is not deleted. Better might be to use ``self._paths['nbd']``